src/inmanta/server/services/databaseservice.py

```diff
@@ -210,58 +226,55 @@ async def connect_database(self) -> None:
             connection_pool_max_size=opt.server_db_connection_pool_max_size.get(),
             connection_timeout=opt.server_db_connection_timeout.get(),
         )
-
-        # Check if JIT is enabled
-        async with self._pool.acquire() as connection:
+        async with get_connection_ctx_mgr() as connection:
+            # Check if JIT is enabled
             jit_available = await connection.fetchval("SELECT pg_jit_available();")
             if jit_available:
                 LOGGER.warning("JIT is enabled in the PostgreSQL database. This might result in poor query performance.")
 
     async def disconnect_database(self) -> None:
         """Disconnect the database"""
-        await data.disconnect()
+        await stop_engine()
 
     async def get_status(self) -> Mapping[str, ArgumentTypes]:
         """Get the status of the database connection"""
         assert self._db_monitor is not None  # make mypy happy
         return (await self._db_monitor.get_status()).dict()
 
 
-async def initialize_database_connection_pool(
+async def initialize_sql_alchemy_engine(
     database_host: str,
     database_port: int,
     database_name: str,
     database_username: str,
     database_password: str,
-    create_db_schema: bool,
-    connection_pool_min_size: int,
-    connection_pool_max_size: int,
-    connection_timeout: float,
-) -> asyncpg.pool.Pool:
+    create_db_schema: bool = False,
+    connection_pool_min_size: int = 10,
+    connection_pool_max_size: int = 10,
+    connection_timeout: float = 60.0,
+) -> None:
     """
-    Initialize the database connection pool for the current process and return it.
+    Initialize the sql alchemy engine for the current process and return it.
 
     :param database_host: Database host address.
     :param database_port: Port number to connect to at the server host.
     :param database_name: Name of the database to connect to.
     :param database_username: Username for database authentication.
     :param database_password: Password for database authentication.
-    :param create_db_schema: Make sure the DB schema is created and up-to-date.
     :param connection_pool_min_size: Initialize the pool with this number of connections .
     :param connection_pool_max_size: Limit the size of the pool to this number of connections .
     :param connection_timeout: Connection timeout (in seconds) when interacting with the database.
     """
 
-    out = await data.connect(
-        host=database_host,
-        port=database_port,
-        database=database_name,
-        username=database_username,
-        password=database_password,
-        create_db_schema=create_db_schema,
-        connection_pool_min_size=connection_pool_min_size,
-        connection_pool_max_size=connection_pool_max_size,
-        connection_timeout=connection_timeout,
+    await start_engine(
+        url=f"postgresql+asyncpg://{database_username}:{database_password}@{database_host}:{database_port}/{database_name}",
```

Comment:
This is probably somewhat injection sensitive, definitely if there are garbage characters in the password. 

We may need to encode the strings (and test that)? 

it seems to be supported: https://docs.sqlalchemy.org/en/20/core/engines.html#creating-urls-programmatically